lib/opam.ml

```diff
@@ -313,23 +313,29 @@ module Url = struct
         else OS.File.must_exist distrib_file >>= fun _ -> assert false
 end
 
-let opam_version () =
-  let v =
+let version v =
+  if Bos_setup.String.is_prefix v ~affix:"1." then (
+    Logs.warn (fun l ->
```

Comment:
I'd rather remove the side effects from this function and deal with the warning in the caller, after the parsing's done.

What do you think?

Reply:
agreed, done.